daemon/daemon.go

```diff
@@ -706,6 +708,15 @@ func NewDaemon(config *Config, registryService registry.Service, containerdRemot
 	// on Windows to dump Go routine stacks
 	d.setupDumpStackTrap(config.Root)
 
+	d.imagePuller = distribution.NewImagePuller(distribution.ImagePullerConfig{
+		RegistryService:  daemon.RegistryService,
+		ImageEventLogger: daemon.LogImageEvent,
+		MetadataStore:    daemon.distributionMetadataStore,
+		ImageStore:       daemon.imageStore,
+		ReferenceStore:   daemon.referenceStore,
+		DownloadManager:  daemon.downloadManager,
```

Comment:
Actually, some of these structs might not need to be in the daemon any more. I'll see if I can remove them.